autotest: footprint test fails if run after mdim convert test

### What is the bug?

Running `autotest/utilities/test_gdalalg_pipeline.py::test_gdalalg_pipeline_footprint` after `autotest/utilities/test_gdalalg_mdim_convert.py::test_gdalalg_mdim_convert_creation_option` fails on OSX. The tests succeed if run independently.

### Steps to reproduce the issue

```
pytest -s autotest/utilities/test_gdalalg_mdim_convert.py::test_gdalalg_mdim_convert_creation_option autotest/utilities/test_gdalalg_pipeline.py::test_gdalalg_pipeline_footprint
```

fails with

```
________________________________________________________________________________________ test_gdalalg_pipeline_footprint _________________________________________________________________________________________

    def test_gdalalg_pipeline_footprint():

>       with gdal.Run(
            "pipeline",
            input="../gcore/data/byte.tif",
            output_format="MEM",
            output="",
            pipeline="read ! footprint ! write",
        ) as alg:

/Users/dan/dev/gdal/build/autotest/utilities/test_gdalalg_pipeline.py:358:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
/Users/dan/dev/gdal/build/swig/python/osgeo/gdal.py:3593: in Run
    if not alg.Run(progress):
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <osgeo.gdal.Algorithm; proxy of <Swig Object of type 'GDALAlgorithmHS *' at 0x1139ce730> >, args = (None,)

    def Run(self, *args):
        r"""Run(self, callback=0, callback_data=None) -> bool"""

        self.has_run = True


>       return _gdal.Algorithm_Run(self, *args)
E       RuntimeError: Terminating translation prematurely after failed
E       translation of layer footprint (use -skipfailures to skip errors)
E       May be caused by: stream.get_next() failed
E       May be caused by: disk I/O error
```

### Versions and provenance

master, 2bde98d96c317e3b1f1fcc79cc3f210431671a57

### Additional context

I'm guessing it has something to do with both tests using `byte.tif`.

--- a/apps/gdalalg_raster_contour.cpp
+++ b/apps/gdalalg_raster_contour.cpp
@@ -264,7 +264,11 @@ bool GDALRasterContourAlgorithm::RunStep(GDALPipelineStepRunContext &ctxt)
             poDstDS->MarkSuppressOnClose();
             if (bRet)
                 bRet = poDstDS->FlushCache() == CE_None;
+#if !defined(__APPLE__)
+            // For some unknown reason, unlinking the file on MacOSX
+            // leads to later "disk I/O error". See https://github.com/OSGeo/gdal/issues/13794
             VSIUnlink(outputFilename.c_str());
+#endif
         }
         m_outputDataset.Set(std::unique_ptr<GDALDataset>(poDstDS));
     }

--- a/apps/gdalalg_raster_footprint.cpp
+++ b/apps/gdalalg_raster_footprint.cpp
@@ -347,7 +347,11 @@ bool GDALRasterFootprintAlgorithm::RunStep(GDALPipelineStepRunContext &ctxt)
             if (poRetDS && !m_standaloneStep && !outputFilename.empty())
             {
                 bOK = poRetDS->FlushCache() == CE_None;
+#if !defined(__APPLE__)
+                // For some unknown reason, unlinking the file on MacOSX
+                // leads to later "disk I/O error". See https://github.com/OSGeo/gdal/issues/13794
                 VSIUnlink(outputFilename.c_str());
+#endif
                 poRetDS->MarkSuppressOnClose();
             }
             m_outputDataset.Set(std::unique_ptr<GDALDataset>(poRetDS));

--- a/apps/gdalalg_raster_polygonize.cpp
+++ b/apps/gdalalg_raster_polygonize.cpp
@@ -284,7 +284,11 @@ bool GDALRasterPolygonizeAlgorithm::RunStep(GDALPipelineStepRunContext &ctxt)
         if (bTemporaryFile)
         {
             ret = poRetDS->FlushCache() == CE_None;
+#if !defined(__APPLE__)
+            // For some unknown reason, unlinking the file on MacOSX
+            // leads to later "disk I/O error". See https://github.com/OSGeo/gdal/issues/13794
             VSIUnlink(outputFilename.c_str());
+#endif
         }
 
         m_outputDataset.Set(std::move(poRetDS));

